src/tree/updater_gpu_hist.cu

```diff
@@ -476,26 +492,34 @@ class GPUHistMaker : public TreeUpdater {
             device_row_segments[d_idx + 1] - device_row_segments[d_idx];
         bst_ulong num_elements_segment =
             device_element_segments[d_idx + 1] - device_element_segments[d_idx];
+
```

Comment:
Bug fix, because the multiplication of int32 easily overflows for these particular items.